ding/torch_utils/network/tests/test_gumbel_softmax.py

```diff
@@ -0,0 +1,19 @@
+import numpy as np
+import pytest
+import torch
+
+from ding.torch_utils.network import GumbelSoftmax
+
+@pytest.mark.unittest
+class TestGumbelSoftmax:
+    def test(self):
+        B = 4
+        N = 10
+        model = GumbelSoftmax()
+        # data case 1
+        for _ in range(N):
+            data = torch.rand((4,10))
+            data = torch.log(data)
+            GumbelSoftmaxDistri = model(data, hard=True)
```

Comment:
add test case for `hard=False`, you can refer to [this](https://github.com/opendilab/DI-engine/blob/main/ding/rl_utils/tests/test_ppo.py#L25)